cpp/bench/ann/src/common/benchmark.hpp

```diff
@@ -38,11 +38,59 @@
 #include <string>
 #include <unistd.h>
 #include <vector>
+
 namespace raft::bench::ann {
 
-std::mutex init_mutex;
-std::condition_variable cond_var;
-std::atomic_int processed_threads{0};
+/**
+ * A simple progress tracker that allows syncing threads multiple times and resets the global
+ * progress once the threads are done.
+ */
+struct progress_sync {
```

Comment:
Can this be moved to `raft_ann_bench_utils.h`?

Reply:
No, because that file is for raft wrappers only. I can move it to the `common/util.hpp` though.